activerecord/test/cases/relation/or_test.rb

```diff
@@ -139,6 +139,13 @@ def test_or_with_scope_on_association
       end
     end
 
+    def test_or_with_annotate
+      assert_match %r{#{Post.quoted_table_name} /\* foo \*/\z},
+        Post.annotate("foo").or(Post.all).to_sql
+      assert_match %r{#{Post.quoted_table_name} /\* foo \*/ /\* bar \*/\z},
+        Post.annotate("foo").or(Post.annotate("bar")).to_sql
```

Comment:
How should `Post.annotate("foo").or(Post.annotate("foo")).to_sql` / `Post.annotate("foo", "bar").or(Post.annotate("foo")).to_sql` behave?
`/* foo */ /* foo */` / `/* foo */ /* bar */ /* foo */`?

Reply:
Thank you for your review.
I have added assertions for those two test cases.

Reply:
Sorry for the late action.
I updated a behavior for backward compatibility and added test codes.